### PR TITLE
Prep for dataset seed script

### PR DESCRIPTION
**Pull request by alexsenden**
# #48 

### Changes: 
- Created a script to insert the dataset into the db
- Made some changes to the db schema to correspond with the dataset structure (sorry Chris, I know I said the IDs should be uuids but they are ints in the dataset and I don't want to mess with that lol)
- Used debouncing for FE call to get songs/albums/artists by partial name to not overload the api

The script is too big to be stored in GitHub, so it's downloadable from OneDrive, linked in the DB README. It's too bad, I really wanted a 65 000 000 line PR.

### CODE REVIEW

File: harmony-api/prisma/schema.prisma
Comment (Dankgen) [inline harmony-api/prisma/schema.prisma:1]: I understand we are likely changing  some Ids into ints due to relations from MusicBrainz, however would it make sense to change the other Ids into ints aswell?

**Comment by Dankgen** [comment]
My one other nitpick is the raw SQL querying. You are more familiar with Prisma than I, however wouldnt these bring SQL injection possibilities to the table?

**Comment by Developik** [comment]
Also, after I merged my branch please add two fields to user object (conflicting files)

File: harmony-api/prisma/schema.prisma
Comment (alexsenden) [reply harmony-api/prisma/schema.prisma:1]: Totally up to you, you make the call. I could see a case for either side.

**Comment by alexsenden** [comment]
> My one other nitpick is the raw SQL querying. You are more familiar with Prisma than I, however wouldnt these bring SQL injection possibilities to the table?

Prisma has bad naming for their raw queries lol, `$queryRaw` does actually use prepared statements under the hood, so it is as resistant to SQL injection as regular Prisma queries are. Good eye though. To actually open it up to SQL injection we would have to be using `$queryRawUnsafe`.

I also get the hesitancy to use raw queries since they are harder to maintain, however this is a request that is going to get run a lot, and on a lot of data, and in my testing against the dataset, this raw query was about 50-100x faster than the regular Prisma query, so I went with it.

**Review by Dankgen — APPROVED** [review]
Thanks for the clarification on the prisma sql query. As for the user uuid, it may just be better in the short term considering most of our code is already set on it being a uuid.
Other than that, you will just have the merge conflict in the schema.prisma file for the new first name and last name fields

**Pull request merged**